WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift

```diff
@@ -0,0 +1,119 @@
+import Yosemite
+/// Orchestrates the sequence of actions required to capture a payment:
+/// 1. Check the there is a card reader connected
+/// 2. Launch the reader discovering and pairing UI if there is no reader connected
+/// 3. Obtain a Payment Intent from the card reader
+/// 4. Submit the Payment Intent to WCPay
+/// Steps 1 and 2 will be implemented as part of https://github.com/woocommerce/woocommerce-ios/issues/4062
+final class PaymentCaptureOrchestrator {
+    private let currencyFormatter = CurrencyFormatter(currencySettings: ServiceLocator.currencySettings)
+
+    func collectPayment(for order: Order,
+                        onPresentMessage: @escaping (String) -> Void,
+                        onClearMessage: @escaping () -> Void,
+                        onCompletion: @escaping (Result<CardPresentReceiptParameters, Error>) -> Void) {
+
+        // TODO. Check that there i a reader currently connected
```

Comment:
This is where we'd check if there is a reader connected, and if there isn't, we'd launch the discovery and pairing UI. For now, we assume there is a reader and attack it directly